internal/clients/sources/sources_impl.go

```diff
@@ -2,12 +2,54 @@ package sources
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
 
 	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
 )
 
+type AppType struct {
+	Id          string `json:"id"`
+	Name        string `json:"name"`
+	DisplayName string `json:"display_name"`
+}
+
+type dataElement struct {
+	Data []AppType `json:"data"`
+}
+
 func getSourcesClient(ctx context.Context) (SourcesIntegration, error) {
-	return NewClientWithResponses(config.Sources.URL)
+	client, err := NewClientWithResponses(config.Sources.URL)
+	if err != nil {
+		return client, err
+	}
+	if config.Sources.AppTypeId == "" {
```

Comment:
well... how bad it is, if we fetch the value multiple times? :thinking: setting should be mutexted, I'd agree, but I feel like it doesn't matter much if we fetch the id 30 times at the beggining ... but let me try to take a look if I can figure something not-too ugly :)

Reply:
what about this? I've just added mutex around fetching the appType, you should not need appTypeId prior having the client, so this should be effectively working as expected...

Reply:
Mutex does not solve the thread-local cache issue unless also read and write is also protected. To do that, the best option is to make the config value unexported and create a "getter" method that will use the mutex. This will ensure that no other goroutine is able to read an invalid (empty/nil) value unless it's been correctly set.

I suggest sync.Once in a "getter" method. That is just like mutex, but cleaner.

Reply:
Yeah, let's do it right, I agree. In that case I'd move that from config completely, or created a specialized structure for it.

IMHO, for this usecase this is fine, because the read doesn't happen until there is a client, but you're right that this code sets an example, so better do it right.

Reply:
Once is not useable, as we need to pass context, so I've used mutex, this is best efort, but happy to talk it over ^_^